experiment/prepare.py

```diff
@@ -15,18 +15,81 @@ def __init__(
         rng: np.random.Generator,
         default_params: Path,
         experiment_params: Path,
+        n_samples: int
```

Comment:
I don't think we need both `n_samples` and `n_iterations` (which is unused now once we get rid of unused functions). At some point you asked about running multiple people with the exact same parameters - I could see this being useful if we wanted to, eg, draw 100 covariate values and simulate each combination of covariate values and scenario 10 times. But I can't think of a reason why we'd want to do that instead of just drawing 1,000 covariate values.

My nomenclature preference between the two is `n_iterations`. Although we use the value under the hood later on to draw samples, the fundamental construct we are defining at experiment instantiation is how many iterations we want to run per scenario.